ambry-protocol/src/main/java/com.github.ambry.protocol/AdminRequest.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.protocol;
+
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.utils.Utils;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.WritableByteChannel;
+
+
+/**
+ * Representation of an administration request. Types of requests are represented by {@link AdminRequestOrResponseType}.
+ */
+public class AdminRequest extends RequestOrResponse {
+  private static final int REQUEST_TYPE_SIZE = 2;
+  private static final short ADMIN_REQUEST_VERSION_V1 = 1;
+
+  private final AdminRequestOrResponseType type;
+  private final PartitionId partitionId;
+  private final long sizeInBytes;
+
+  private int sizeSent = 0;
+
+  /**
+   * Reads from a stream and constructs an {@link AdminRequest}.
+   * @param stream the {@link DataInputStream} to read from.
+   * @param clusterMap the {@link ClusterMap} in use.
+   * @return {@link AdminRequest} that is deserialized from the {@code stream}.
+   * @throws IOException if there is an I/O error reading from {@code stream}
+   */
+  public static AdminRequest readFrom(DataInputStream stream, ClusterMap clusterMap) throws IOException {
+    Short versionId = stream.readShort();
+    int correlationId = stream.readInt();
+    String clientId = Utils.readIntString(stream);
+    AdminRequestOrResponseType type = AdminRequestOrResponseType.values()[stream.readShort()];
+    AdminRequest request;
+    switch (type) {
+      case TriggerCompaction:
+        PartitionId id = clusterMap.getPartitionIdFromStream(stream);
+        request = new AdminRequest(type, id, correlationId, clientId);
+        break;
+      default:
+        throw new IllegalArgumentException("Unrecognized admin request type: " + type);
+    }
+    // ignore version for now
```

Comment:
Why is this comment down here? Also, I would say just handle the version (even if it is something as simple as `if (version != V1) throw ...`).

I just say this because we had to be careful before when we did not check the version for `PutRequests` and we wanted to create a new version. We always have to follow the procedure of first having the servers understand the new version and then have the frontends write in the new version - but with the version checks already in place, the worst that could happen was that requests would fail (if the frontends were made to write in the new version right away). If version checks are not already in place, the requests can get processed successfully but incorrectly.